src/main/java/com/github/dockerjava/api/command/DockerCmdExecFactory.java

```diff
@@ -105,6 +107,8 @@
 
     public DisconnectFromNetworkCmd.Exec createDisconnectFromNetworkCmdExec();
 
+    public DockerCmdExecFactory withSSLContext(SSLContext sslContext);
```

Comment:
Passing a custom `SSLContext` to the `DockerClientConfig` could logically collide with the DOCKER_TLS_VERIFY/dockerTlsVerify() option in the sense of consistency. Additionally I want to free `DockerClientConfig` from non-default/special configuration options so I moved it to the `DockerCmdExecFactory`. Makes sense?